### ISSUE TITLE
SoftAP command: Misunderstanding of the length of device public key 

### ISSUE THREAD

**Issue by XuGuohui** [spark/firmware#1083]
Hi there,

In file "softap.cpp":

```
const int length = EXTERNAL_FLASH_SERVER_PUBLIC_KEY_LENGTH;
const uint8_t* data = fetch_device_public_key();
```

The macro `EXTERNAL_FLASH_SERVER_PUBLIC_KEY_LENGTH` is used as the length of the device public key to be fetched. 

Is it a typo?

**Comment by technobly** [comment]
Yes that does look like a typo :)  Should probably be `DCT_DEVICE_PUBLIC_KEY_SIZE` which is 384, instead of `EXTERNAL_FLASH_SERVER_PUBLIC_KEY_LENGTH` which is 294.  It would short change the returned device public key.
